x/commerciodocs/keeper.go

```diff
@@ -127,22 +127,23 @@ func (keeper Keeper) ShareDocument(ctx sdk.Context, reference string, sender typ
 	sharingStore := ctx.KVStore(keeper.sharingStoreKey)
 
 	// Save the shared document for the sender
-	if !keeper.CanReadDocument(ctx, sender, reference) {
+	if keeper.CanReadDocument(ctx, sender, reference) {
 		sharingStore.Set([]byte(sender), keeper.cdc.MustMarshalBinaryBare(sharing))
+	} else {
+		return sdk.ErrUnauthorized(fmt.Sprintf("The sender with address %s doesnt have the rights on this document", sender))
 	}
 
 	// Save the shared document for the recipient
 	if !keeper.CanReadDocument(ctx, recipient, reference) {
 		sharingStore.Set([]byte(recipient), keeper.cdc.MustMarshalBinaryBare(sharing))
+		keeper.addReaderForDocument(ctx, recipient, reference)
 	}
 
-	// Set both the users as readers
-	keeper.addReaderForDocument(ctx, sender, reference)
-	keeper.addReaderForDocument(ctx, recipient, reference)
+	return nil
 }
 
-// CanReadDocument tells whenever a given user has access to a document or not.
-// Returns true if the user has access to the document, false otherwise.
+// CanReadDocument tells whenever a given ownerIdentity has access to a document or not.
+// Returns true if the ownerIdentity has access to the document, false otherwise.
```

Comment:
I think both of the `ownerIdentity` instances should be brought back to `user` to make it more clear overall.